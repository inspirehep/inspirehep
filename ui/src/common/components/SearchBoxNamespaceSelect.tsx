import React from 'react';

import SelectBox from './SelectBox';
import { DATA_NS, SEARCH_BOX_NAMESPACES } from '../../search/constants';

const SearchBoxNamespaceSelect = ({
  canAccessDataCollection,
  searchScopeName,
  onSearchScopeChange,
}: {
  canAccessDataCollection: boolean;
  searchScopeName: string;
  onSearchScopeChange: Function;
}) => {
  const filteredNamespaces = canAccessDataCollection
    ? SEARCH_BOX_NAMESPACES
    : SEARCH_BOX_NAMESPACES.filter((value) => value !== DATA_NS);
  const SCOPE_OPTIONS = filteredNamespaces.map((value) => ({ value }));

  return (
    <SelectBox
      popupClassName="header-dropdown"
      onChange={onSearchScopeChange}
      value={searchScopeName}
      options={SCOPE_OPTIONS}
    />
  );
};

export default SearchBoxNamespaceSelect;
