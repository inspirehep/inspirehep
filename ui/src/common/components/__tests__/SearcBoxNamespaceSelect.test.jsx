import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import SearchBoxNamespaceSelect from '../SearchBoxNamespaceSelect';

describe('SearchBoxNamespaceSelect', () => {
  it('render initial state with all props set', () => {
    const { asFragment } = render(
      <SearchBoxNamespaceSelect
        onSearchScopeChange={jest.fn()}
        searchScopeName="authors"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onSearchScopeChange on select change', () => {
    const onSearchScopeChange = jest.fn();
    const { getByRole, getByText } = render(
      <SearchBoxNamespaceSelect
        searchScopeName="literature"
        onSearchScopeChange={onSearchScopeChange}
      />
    );

    const selectBox = getByRole('combobox');
    fireEvent.mouseDown(selectBox);
    const newScopeOption = getByText('jobs');
    fireEvent.click(newScopeOption);
    expect(onSearchScopeChange).toBeCalledWith('jobs', expect.any(Object));
  });
});
