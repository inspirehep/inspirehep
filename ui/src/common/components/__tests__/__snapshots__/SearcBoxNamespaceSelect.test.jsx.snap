// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBoxNamespaceSelect render initial state with all props set 1`] = `
<SelectBox
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "value": "literature",
      },
      Object {
        "value": "authors",
      },
      Object {
        "value": "jobs",
      },
      Object {
        "value": "seminars",
      },
      Object {
        "value": "conferences",
      },
      Object {
        "value": "institutions",
      },
      Object {
        "value": "experiments",
      },
      Object {
        "value": "journals",
      },
      Object {
        "value": "data",
      },
    ]
  }
  popupClassName="header-dropdown"
  value="authors"
/>
`;
